### ISSUE TITLE
tooltips on touch device

### ISSUE THREAD

**Issue by gabrieldodan** [seiyria/bootstrap-slider#513]
Hi

Tooltips are not shown on touch devices. Is there any option or this feature is not implemented? Thanks.

**Comment by seiyria** [comment]
set `tooltip` to `always`.

**Comment by gabrieldodan** [comment]
Yes but in this case the tooltip will be shown always. I would like to be shown only when user touch the handler or the bar. Probably i will have to implement in other way. Thanks anyway.

**Comment by kobusvanwyk** [comment]
Possible solution is to add "touchstart", "touchmove" and "touchend" to the handle event listener.

**From line 713:**
`

				this.sliderElem.addEventListener("mouseenter", this.showTooltip, false);
				this.sliderElem.addEventListener("mouseleave", this.hideTooltip, false);

				this.handle1.addEventListener("focus", this.showTooltip, false);
				this.handle1.addEventListener("blur", this.hideTooltip, false);

				this.handle1.addEventListener("touchstart", this.showTooltip, false);
				this.handle1.addEventListener("touchmove", this.showTooltip, false);
				this.handle1.addEventListener("touchend", this.hideTooltip, false);

				this.handle2.addEventListener("focus", this.showTooltip, false);
				this.handle2.addEventListener("blur", this.hideTooltip, false)

				this.handle2.addEventListener("touchstart", this.showTooltip, false);
				this.handle2.addEventListener("touchmove", this.showTooltip, false);
				this.handle2.addEventListener("touchend", this.hideTooltip, false);
`

**Comment by Fenny** [comment]
@kobusvanwyk 
This solves the problem!

But it throws an error in chrome:

'[Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952'
